### DISCUSSION TITLE
Global packageRules ?

### DISCUSSION THREAD

**Discussion by sjoerd-dijkstra** [renovatebot/renovate#17813] [Request Help] [answered]
### Question

We have defined a number of packageRules that we are using in many (if not all) our individual repositories' `renovate.json` file. Rather than copy-paste the same config, is there a way to configure these rules only once globally? (see #17806 for reference)

My assumption would be to use a modified default renovate scheme, am i right?

We're using self hosted gitlab renovate runners.

Thank in advance! 🙏

**Comment by HonkingGoose** [comment, marked as answer]
What you want is called a "shareable config preset".

The basic concept is (do not use this a guide, read the docs 😉):

1. Create a config file (`default.json`) that you host on a _separate_ repository
2. Put the copy/paste `packageRules` in the `default.json`
3. In each of your individual repositories, edit your `renovate.json` file:
   - `extend` the new global config
   - remove the old copy/pasted config your `renovate.json` files in each repository
5. Profit! 💰

Take a look a the [Renovate docs, Shareable Config Presets](https://docs.renovatebot.com/config-presets/) for the correct instructions.

**Reply by sjoerd-dijkstra** [reply]
Thank you! This is exactly what we were looking for.

We now have defined a sharable config `default.json` file in our renovate repo, on which we extend in the individual repos.

When onboarding new projects, this config is now also suggested via `RENOVATE_ONBOARDING_CONFIG`.

Nice! 😎🙏

**Comment by rarkins** [comment]
Shared presets as recommended by @HonkingGoose are the best because it provides visibility/transparency for the downstream users. 

Alternatively if you self host you can configure on the "bot" side but then users in the repo won't really have visibility into it.
